src/components/TradeWidget/index.tsx

```diff
@@ -301,10 +301,10 @@ const TradeWidget: React.FC<TradeWidgetProps> = ({
   useEffect(() => {
     if (priceValue && sellValue) {
       // If price is quoted in sell tokens, we use it, otherwise we use the inverse
-      const priceUsedForReceiveAmount = sellToken === quoteToken ? priceValue : priceInverseValue
+      const priceUsedForReceiveAmount = sellToken.address === quoteToken.address ? priceValue : priceInverseValue
```

Comment:
is this related to the issue? was the problem related to "identity"?

Reply:
I made this change as I was seeing a lot of re-renders in this hook, not sure if the references between the objects `sellToken` and `quoteToken` is stable. anyways can remove